File: source-connector/src/main/java/sourceconnector/config/StorageRepositoryConfiguration.java
package sourceconnector.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Import;
import sourceconnector.repository.file.validator.FileValidator;

@Import(StorageRepositoryRegistrar.class)
@Configuration
public class StorageRepositoryConfiguration {
  @Bean
  public FileValidator fileValidator(FileSearchConfigs fileSearchConfigs) {
    return fileSearchConfigs.toValidator();
  }

}

File: source-connector/src/main/kotlin/sourceconnector/config/StorageRepositoryConfiguration.kt
package sourceconnector.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Import
import sourceconnector.repository.file.validator.FileValidator

@Import(StorageRepositoryRegistrar::class)
@Configuration
class StorageRepositoryConfiguration {
  @Bean
  fun fileValidator(fileSearchConfigs: FileSearchConfigs): FileValidator {
    return fileSearchConfigs.toValidator()
  }
}
